Bjorncs/vespa enforcer extensions

bjorncs opened the pull request:
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

bjorncs committed:
Add custom rule for verifying transitive Maven dependencies

bjorncs committed:
Use custom maven-enforcer-plugin rule in cloud-tenant-base-dependencies-enforcer

bjorncs committed:
Remove rules from dependencies no longer present

gjoranv commented on vespa-enforcer-extensions/src/main/java/com/yahoo/vespa/maven/plugin/enforcer/DependencyEnforcer.java:
```diff
@@ -0,0 +1,150 @@
+ *
+ * @author bjorncs
+ */
+public class DependencyEnforcer implements EnforcerRule {
```
Suggest to rename to _EnforceDependencies_ for consistency with most existing enforcer rules.. Rule name should also be _enforceDependencies_. Where is the actual rule name defined, is it just a non-capitalized version of the class name?

bjorncs replied:
Yes, there is some reflection magic under the hood. That is how injection of the `allowed` list is done.

bjorncs committed:
Misc improvements

Rename 'DependencyEnforcer' => 'EnforceDependencies'.
Improve javadoc. Add unit tests. Log on successful completion.

bjorncs committed:
Use custom maven-enforcer-plugin rule in container-dependencies-enforcer

bjorncs commented:
@gjoranv PTAL

gjoranv reviewed (approved):
This is great!
